pkg/flavor/kubernetes/multiwriter.go

```diff
@@ -0,0 +1,673 @@
+		return nil
+	}
+
+	// create the nfs service
+	service := flavor.makeNFSService(svcName)
+	if _, err := flavor.kubeClient.CoreV1().Services(multiWriterNamespace).Create(service); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create nfs service %s, err %+v", svcName, err)
+		}
+		log.Infof("nfs service %s already exists", svcName)
+	} else {
+		log.Infof("nfs service %s started", svcName)
+	}
+	return nil
+}
+
+func (flavor *Flavor) GetNFSService(svcName string) (*core_v1.Service, error) {
+	log.Tracef(">>>>> GetNFSService with service name %s", svcName)
+	defer log.Tracef("<<<<< GetNFSService")
+
+	service, err := flavor.kubeClient.CoreV1().Services(multiWriterNamespace).Get(svcName, meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// CreateNFSDeployment creates a nfs deployment with given name
+func (flavor *Flavor) CreateNFSDeployment(deploymentName, pvcName string) error {
+	log.Tracef(">>>>> CreateNFSDeployment with name %s", deploymentName)
+	defer log.Tracef("<<<<< CreateNFSDeployment")
+
+	// validate nfs deployment settings
+	if deploymentName == "" {
+		return fmt.Errorf("empty name provided for creating NFS multiwriter deployment")
+	}
+
+	// check if nfs deployment already exists
+	getAction := func() error {
+		_, err := flavor.kubeClient.AppsV1().Deployments(multiWriterNamespace).Get(deploymentName, meta_v1.GetOptions{})
+		return err
+	}
+	exists, err := flavor.resourceExists(getAction, "deployment", deploymentName)
+	if err == nil && exists {
+		log.Infof("nfs deployment %s exists in %s namespace", deploymentName, multiWriterNamespace)
+		return nil
+	}
+
+	// create a nfs deployment
+	deployment := flavor.makeNFSDeployment(deploymentName, pvcName)
+	if _, err := flavor.kubeClient.AppsV1().Deployments(multiWriterNamespace).Create(deployment); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create nfs deployment %s, err %+v", deploymentName, err)
+		}
+		log.Infof("nfs deployment %s already exists", deploymentName)
+		return nil
+	}
+	// make sure its available
+	err = flavor.waitForDeployment(deploymentName)
+	if err != nil {
+		return err
+	}
+	// successfully created nfs deployment. make sure its running.
+	log.Infof("nfs deployment %s started", deploymentName)
+
+	return nil
+}
+
+//nolint
+func (flavor *Flavor) makeNFSService(svcName string) *core_v1.Service {
+	log.Tracef(">>>>> makeNFSService with name %s", svcName)
+	defer log.Tracef("<<<<< makeNFSService")
+
+	labels := map[string]string{"app": getDeploymentFromServiceName(svcName)}
+	svc := &core_v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      svcName,
+			Namespace: multiWriterNamespace,
+			Labels:    labels,
+		},
+		Spec: core_v1.ServiceSpec{
+			Selector: labels,
+			Type:     core_v1.ServiceTypeClusterIP,
+			Ports: []core_v1.ServicePort{
+				{Name: "grpc", Port: 49000, Protocol: core_v1.ProtocolTCP},
```

Comment:
Cool.  Do any of these need to be configurable?

Reply:
these are standard i guess for each nfs-pod. will investigate more..